discord/message.py

```diff
@@ -988,6 +988,21 @@ def is_system(self) -> bool:
             MessageType.application_command,
             MessageType.thread_starter_message,
         )
+    
+    @property
+    def thread(self) -> Optional[Thread]:
+        """Retrieves the thread the message has created ``None`` if no thread was created
+        with this message or if the channel is not a :class:`TextChannel`.
+        
+        Returns
+        --------
+        Optional[:class:`Thread`]
+            The thread this message has created.
+        """
+        try:
+            return self.channel.get_thread(self.id)
+        except AttributeError:
+            return None
```

Comment:
`get_thread` should return `None` if not found. The exception handler isn't needed.

Reply:
It is, if the channel is DMChannel or something other than a text channel, it does not have the attribute get_thread. In which case, we will return None.

Reply:
Right, gotcha! In that case, I'd personally check the channel type before hand and return early. Generally not a huge fan of leaning on exceptions for control flow.